onecodex/viz/_bargraph.py

```diff
@@ -81,6 +81,12 @@ def plot_bargraph(
         if not (threshold or top_n):
             raise OneCodexException("Please specify at least one of: threshold, top_n")
 
+        if len(self._results) < 1:
```

Comment:
Let's change this to be after line 98 (`df = self.to_df(...)`) so that we catch instances where there are no samples left after we filter for taxa.

Reply:
I couldn't get `to_df()` to drop samples during taxa filtering. Even if all taxa are filtered out, the returned `df` contains the input samples.

Also, I found that `to_df()` raises a `OneCodexException` if the sample collection is empty, so I don't think we can put the check after `to_df()` because it won't be reached before the other error is raised.

Do you think it makes sense to leave this check where it's at?